recipes/open62541/all/test_package/conanfile.py

```diff
@@ -1,25 +1,35 @@
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake, CMakeToolchain
 import os
 
 
+# It will become the standard on Conan 2.x
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package_multi"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
+    test_type = "explicit"
 
     def requirements(self):
+        self.requires(self.tested_reference_str)
         self.requires("ua-nodeset/padim-1.02-2021-07-21")
 
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["open62541_NODESET_DIR"] = self.deps_user_info["ua-nodeset"].nodeset_dir
+        tc.variables["open62541_TOOLS_DIR"] = self.dependencies["open62541"].conf_info.get(
+            "user.open62541:tools_dir")
```

Comment:
```suggestion
        tc.variables["open62541_TOOLS_DIR"] = self.dependencies["open62541"].conf_info.get(
            "user.open62541:tools_dir").replace("\\", "/")
```

it's your responsibility to pass valid paths to CMake variables.

Reply:
good to know, however this was not the case with the v1

Reply:
Indeed, but I think it's better, it avoids unwanted conversions.

Reply:
@SpaceIm should this be checked with conan hooks or the linter?